src/concepts/airflow_gotchas.md

```diff
@@ -0,0 +1,72 @@
+# Airflow Gotcha's
+
+Airflow is an integral part of our data platform. ETL processes, forecasts and various analyses are scheduled and monitored through Airflow. Our Airflow instance is hosted at [`workflow.telemetry.mozilla.org`](https://workflow.telemetry.mozilla.org/home) (WTMO).
+
+## DAGs are automatically generated for the most part
+
+Airflow DAGs for our ETL processes get automatically generated as part of [bigquery-etl](https://github.com/mozilla/bigquery-etl). The process for scheduling queries is to specify a DAG as part of its metadata. How to schedule queries is described in detail this [guide to creating a derived dataset](https://mozilla.github.io/bigquery-etl/cookbooks/creating_a_derived_dataset/). Generated DAGs are prefixed with `bqetl_`.
+
+Some DAGs, for example for custom machine learning tasks or to schedule running custom tools, need to be defined manually. These DAGs need to be created in the [telemetry-airflow](https://github.com/mozilla/telemetry-airflow/tree/main/dags) repository.
+
+A separate script syncs generated bigquery-etl DAGs every 10 minutes to our Airflow instance. DAGs that live in telemetry-airflow get deployed via CircleCI whenever a change is pushed to `main`.
+
+## New DAGs need to be unpaused manually
+
+After adding a new DAG either through bigquery-etl or telemetry-airflow, it will take about 10 minutes until the new DAG gets deployed. After deployment, the DAG is by default disabled. It is necessary to manually _unpause_ the DAG on [WTMO](https://workflow.telemetry.mozilla.org/home)
+
+## External task dependencies are managed via `ExternalTaskSensor`s
+
+Tasks are distributed across different Airflow DAGs. Usually, each DAG contains tasks that are closely related to a specific use case or for generating a set of related datasets. In many cases, tasks depend on other tasks that are running as part of a different DAG. For example, a lot of tasks depend on the [`copy_deduplicate_main_ping`](https://github.com/mozilla/telemetry-airflow/blob/0ba2b5631f079fa90fe07467021fab0f9cfc7366/dags/copy_deduplicate.py#L116) task.
+
+External upstream dependencies are expressed using [`ExternalTaskSensor`s](https://airflow.apache.org/docs/apache-airflow/1.10.3/_api/airflow/sensors/external_task_sensor/index.html). These sensors ensure that the external upstream task is finished before the job that depends on that upstream task is executed. These sensors are usually defined like:
+
+```python
+wait_for_bq_events = ExternalTaskSensor(
+    task_id="wait_for_bq_events",       # name of this wait task as it will appear in the UI
+    external_dag_id="copy_deduplicate", # name of the external DAG
+    external_task_id="bq_main_events",  # name of the external task
+    execution_delta=timedelta(hours=3), # delta based on differences in schedule between upstream DAG and current DAG
+    mode="reschedule",                  # use mode "reschedule" to unblock slots while waiting on upstream task to finish
```

Comment:
Does it make sense to add a link to other possible options or is this the expected setup for all cases?

Reply:
This is the setup we probably use in 90% of all cases. The `ExternalTaskSensor` in the paragraph above links to the Airflow docs in case people need to look up more information or config options.